core/src/test/java/org/apache/iceberg/TestSequenceNumberForV2Table.java

```diff
@@ -0,0 +1,344 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg;
+
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+import org.apache.iceberg.exceptions.CommitFailedException;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestSequenceNumberForV2Table extends TableTestBase {
+
+  public TestSequenceNumberForV2Table() {
+    super(2);
+  }
+
+  @Test
+  public void testFastAppend() throws IOException {
```

Comment:
The motivation for #1038 was to cover more cases for each action. Instead of having one test case for `FastAppend`, we should add the necessary assertions to all `FastAppend` tests. That keeps the tests in one place and increases coverage of situations that validate sequence numbers.

Could you make sure that the cases that you're testing here are covered by assertions in the `FastAppend` tests so we don't need this case?

Ideally, I'd prefer to add assertions to all of the operation tests instead of this suite, but for now I think it would be reasonable to just update the cases in `FastAppend` and add the rest of this suite. We can replace test cases here with assertions in the right test suites over a few more PRs.

Reply:
This case could be coved by `testEmptyTableAppend` and `testEmptyTableAppendManifest`. Let me remove this.

The idea to move these unit tests to each operation test respectively sounds reasonable to me.  I will try to create some following PRs to clean this.